### PR TITLE
Add maven3 snaphots and sjs support

### PR DESCRIPTION
**Pull request by InTheNow**
Some notes:

- The encrypted passwords in the travis file need setting as per [this cakesloutions guide](http://www.cakesolutions.net/teamblogs/publishing-artefacts-to-oss-sonatype-nexus-using-sbt-and-travis-ci)
- The scalajs code compiles but will not work until all dependent libraries are all available in sjs, especially scalatest
- I have tested using my nexus, but not the cats one
- Travis automation is __completely__ untested

### CODE REVIEW

**Comment by non** [comment]
Can you retarget this PR toward a topic branch? That way, I can merge it now, add the passwords, and prepare for a point where we can merge it into master.

(I'm assuming we shouldn't merge this to master now, since the dependent libraries are not all available.)

File: .travis.yml [hunk 1]
```diff
@@ -10,4 +10,10 @@ notifications:
     on_success: change  # options: [always|never|change] default: always
     on_failure: always  # options: [always|never|change] default: always
     on_start: false     # default: false
+env:
+  global:
+  - secure: STYk3FPnQwBHT25T.../O2SWbVYIL74K...XQAw=
+  - secure: MwRqcAgpBGrmyRY6.../lB/sjDhMk67Y...Xjgk=
 
+after_success:
+  - "[[ $TRAVIS_BRANCH == \"master\" ]] && { sbt publish; };"
```
Comment (dwijnand) [inline .travis.yml]: Shamelessly [taking from Shapeless], we might want a `"${TRAVIS_PULL_REQUEST}" == "false"` here at least.

  [taking from Shapeless]: https://github.com/milessabin/shapeless/blob/a1baba9acb06d6be63b501656ee3a851da9b3244/.travis.yml#L15

**Comment by InTheNow** [comment]
Just a quick note on the failing build. To recreate locally I had to `rm -fr ~/.ivy2/ ~/.m2/` to replicate the issue.

I only mention this as an example of when using the travis cache directories (as in shapeless above) is detrimental as in this case the build would probably have succeeded - and on current  devs machines too. Then, some time in the future, a new dev and/or an existing dev cleans the cache and the build fails...

File: .travis.yml (hunk 1 above)
Comment (InTheNow) [inline .travis.yml]: @dwijnand  thanks for spotting this :)

As luck would have it, @milessabin also updated [the shapeless script](https://github.com/milessabin/shapeless/blob/master/.travis.yml) earlier so I shall base the change on that and trust that this citation to his prior art is sufficient ;)

**Comment by InTheNow** [comment]
> Can you retarget this PR toward a topic branch

Yes, but in a new PR

> (I'm assuming we shouldn't merge this to master now, since the dependent libraries are not all available.)

Actually we can ( and in my view should). As we do not the sjs dep libs, we cannot test the code and hence we cannot release the code. But the code can be compiled with the sjs compiler, and does so successfully. 

So, I have set the build to compile, but not test nor publish sjs projects. This way can ease sjs into cats in a more comfortable fashion . What do you think?

**Comment by InTheNow** [comment]
@non If you would still like me to do a new PR to a new branch, could you create it for me please? eg topic/sjs as you suggested

**Comment by non** [comment]
@InTheNow alright, branch created.

**Comment by InTheNow** [comment]
Superseded by  https://github.com/non/cats/pull/326

**Pull request closed without merging**